controllers/groupsync_controller.go

```diff
@@ -163,7 +167,7 @@ func (r *GroupSyncReconciler) Reconcile(context context.Context, req ctrl.Reques
 
 			if err != nil {
 				log.Error(err, "Failed to Create or Update OpenShift Group")
-				return r.ManageError(context, instance, err)
+				return r.wrapMetricsErrorWithMetrics(prometheusLabels, context, instance, err)
```

Comment:
Is this the right approach long term? Just asking to see if its'something we can put in operator-utils.
Basically why are you associating the outcome with updating the metrics or should it be a separate utility? 
Also from the code it looks like you don't do it on successful runs.

Reply:
I am using that function as a simplified way to manage things that need to be accounted for when an error occurs.

We can certainly it as a further discussion point on the management of metrics as a separate utility

Successful sync metrics are set here:

https://github.com/redhat-cop/group-sync-operator/pull/76/files#diff-5fb0ac19144bbff704ddef730f7782714e92864070261989298c1ced78ddce93R179